src/core/lib/iomgr/tcp_posix.cc

```diff
@@ -381,10 +385,15 @@ class TcpZerocopySendCtx {
   //
   // Please refer to the STATE TRANSITION DIAGRAM below for more details.
   //
-  bool UpdateZeroCopyOMemStateAfterSend(bool seen_enobuf) {
+  bool UpdateZeroCopyOMemStateAfterSend(bool seen_enobuf, bool& cont) {
     MutexLock guard(&lock_);
     is_in_write_ = false;
+    cont = false;
     if (seen_enobuf) {
+      if (ctx_lookup_.size() == 1) {
+        // There is no un-acked z-copy record. Retry the sendmsg again.
```

Comment:
I wouldn't use acked in this context since it might be confused for TCP Acks, which this is not. Also, say why we are retrying the sendmsg().

Something like: "There are no other pending TCP Tx0cp completions, and thus we will not receive Tx0cp EPOLLERR events since this sendmsg() failed. Therefore, set the socket as writeable now so we retry the send in a timely manner."